src/barcodes/codabar/index.js

```diff
@@ -0,0 +1,56 @@
+// Encoding specification:
+// http://www.barcodeisland.com/codabar.phtml
+
+class codabar{
+	constructor(string){
+		this.string = string.toUpperCase();
+
+		if (this.string.search(/^[0-9\-\$\:\.\+\/]*$/) === 0) {
+			this.string = "A" + this.string + "A";
+		}
+
+		this.encodings = {
+			"0": "101010011",
+			"1": "101011001",
+			"2": "101001011",
+			"3": "110010101",
+			"4": "101101001",
+			"5": "110101001",
+			"6": "100101011",
+			"7": "100101101",
+			"8": "100110101",
+			"9": "110100101",
+			"-": "101001101",
+			"$": "101100101",
+			":": "1101011011",
+			"/": "1101101011",
+			".": "1101101101",
+			"+": "101100110011",
+			"A": "1011001001",
+			"B": "1010010011",
+			"C": "1001001011",
+			"D": "1010011001"
+		};
+	}
+
+	valid(){
+		return this.string.search(/^[A-D]?[0-9\-\$\:\.\+\/]*[A-D]$/) !== -1;
```

Comment:
The start and stop characters are marked option in order to allow a "naked" string (e.g. one without them) to be passed in. If that is the case, then the encoder assumes to add "A" as both the start and stop (see the 
The start and stop characters are marked option in order to allow a "naked" string (e.g. one without them) to be passed in. If that is the case, then the encoder assumes to add "A" as both the start and stop (see line 8). I guess I'm not clear on exactly where valid() gets called.

Two other Codabar generators I came across (https://www.nayuki.io/res/1d-barcode-generator-javascript/1d-barcode-generator.js, http://www.jqueryscript.net/other/Simple-jQuery-Based-Barcode-Generator-Barcode.html) both always tack on "A" as start and stop, but the specification allows for arbitrary start/stop within the A-D range. I thought this would be a good middle ground: use "A" if you don't know any better, but keep user-submitted valid start/stops.